### PR TITLE
Configure monitor module mnrva 738

### PR DESCRIPTION
**Pull request by pras7430**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-738

 ## Description:

Monitor route added in the admin dashboard

 ## Testing:
npm run test:admin
 ## Screenshots:
Screenshot 2021-04-27 at 8 23 56 PM

(if applicable)

### CODE REVIEW

File: projects/admin/src/app/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -1,37 +1,50 @@
...
-          )
-          .subscribe();  
-        }
-    });
-   }
+export class DashboardComponent implements AfterViewInit{
+  @ViewChild('tab1Content', { read: ViewContainerRef }) dynamicTabPlaceholder;
+  @ViewChild('tab2Content', { read: ViewContainerRef }) dynamicTabPlaceholder2;
+  @ViewChild('tab3Content', { read: ViewContainerRef }) dynamicTabPlaceholder3;
+  foucsedViewContainer: ViewContainerRef;
```
Comment (lovedeep92) [inline projects/admin/src/app/dashboard/dashboard.component.ts:19]: can you please correct the spelling  **foucsedViewContainer** here !

File: projects/admin/src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
 
-   myfunc() {
-     console.log("inside my func");
-   }
+  constructor(private dataSer: DataService) {}
+
+  ngAfterViewInit(){
+    this.dataSer.getComponentName().subscribe(messageSource => {
+      if (messageSource) {
+        this.dataSer.loadComponent(this.foucsedViewContainer?this.foucsedViewContainer:this.dynamicTabPlaceholder, messageSource);
```
Comment (lovedeep92) [inline projects/admin/src/app/dashboard/dashboard.component.ts:27]: same here!

**Review by lovedeep92 — COMMENTED** [review]
hey I am curious, how Circular JSON issue for Monitors got fixed for you?

File: src/app/_shared/utils.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ const mergeUniqueObjectsOfArray = (original, newdata, selector = 'key') => {
  * @param router Router
  */
 const isAdmin = (router: Router | any): boolean => {
-    return router.url.includes('/admin');
+    return router.url.includes('/dashboard');
```
Comment (FarajiA) [inline src/app/_shared/utils.ts:46]: Nice. This takes care of https://jira.rax.io/browse/MNRVA-760

File: projects/admin/src/app/app.module.ts [hunk 4]
```diff
@@ -39,17 +31,50 @@ import { AngularFireModule } from '@angular/fire';
 import { AngularFireAuthModule } from '@angular/fire/auth';
 import { DashboardComponent } from 'projects/admin/src/app/dashboard/dashboard.component';
 import { DataService } from './_services/data.service';
+import { AJV_INSTANCE, SchemaService } from 'src/app/_services/monitors/schema.service';
+import ajv from 'ajv';
+import { DetailsComponent } from './dashboard/_features/monitors/pages/details/details.component';
```
Comment (FarajiA) [inline projects/admin/src/app/app.module.ts:36]: Hey I'm just honing in on this now but the levels of folders here seems a lot. Didn't wanna just make the change but I think it should be just `/monitors/monitor.details.component` I don't think we have other views for these features.

**Comment by FarajiA** [comment]
Err... I made some updates but it was just to my original naming of the files and class names.

**Review by FarajiA — APPROVED** [review]

File: projects/admin/src/app/app.module.ts (hunk 4 above)
Comment (pras7430) [reply projects/admin/src/app/app.module.ts]: Yeah Right!, I added those which are necessary for monitors. I'll try to remove others.

File: projects/admin/src/app/dashboard/dashboard.component.ts (hunk 1 above)
Comment (pras7430) [reply projects/admin/src/app/dashboard/dashboard.component.ts:19]: Good Catch!

**Comment by pras7430** [comment]
> hey @pras7430 , I am curious, how Circular JSON issue for Monitors got fixed for you?

in my case it was due to not importing the { AJV_INSTANCE, SchemaService } and AJV class and it was not allowed to subscribe through templet, and ivy disable helped me to debug it.

**Pull request merged**